### ISSUE TITLE
Server stops all the time when running targets due to "Stopping the daemon the set of ignored files changed (native)"

### ISSUE THREAD

**Issue by Oidlichtnwoada** [nrwl/nx#20246]
### Current Behavior

Server stops when I execute my two targets:
npx nx run model-import-api:sls-deploy
npx nx run energy-assistant-b2c:serve

The server does not stop when run with --skip-nx-cache in both cases, I am using the local runner, not the cloud runner

PS C:\Users\hanne\development\energy-assistant> npx nx run huawei-import-api:sls-deploy                

   ⠴    Nx is waiting on 4 dependent project tasks before running tasks from huawei-import-api...

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————— 

 >  NX   Ran target sls-deploy for project huawei-import-api and 6 task(s) they depend on (2s)

    ×    0/0 failed
    √    0/0 succeeded [0 read from cache]

 >  NX   Daemon process terminated and closed the connection

   Please rerun the command, which will restart the daemon.
   If you get this error again, check for any errors in the daemon process logs found in: C:\Users\hanne\development\energy-assistant\.nx\cache\d\daemon.log
   
   
  Logs:
  [NX Daemon Server] - 2023-11-15T11:12:52.785Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:12:52.975Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:14:33.093Z - [REQUEST]: Responding to the client. recordOutputsHash
[NX Daemon Server] - 2023-11-15T11:14:33.094Z - Done responding to the client recordOutputsHash
[NX Daemon Server] - 2023-11-15T11:14:33.094Z - Handled RECORD_OUTPUTS_HASH. Handling time: 0. Response time: 1.
[NX Daemon Server] - 2023-11-15T11:14:33.105Z - Closed a connection. Number of open connections: 0
[NX Daemon Server] - 2023-11-15T11:14:33.143Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:16:02.131Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:21:51.258Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:22:05.432Z - [WATCHER]: apps/import-scheduler/serverless.yml was modified
[NX Daemon Server] - 2023-11-15T11:22:05.434Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:22:05.669Z - [REQUEST]: Updated file-hasher based on watched changes, recomputing project graph...
[NX Daemon Server] - 2023-11-15T11:22:05.669Z - [REQUEST]: apps/import-scheduler/serverless.yml
[NX Daemon Server] - 2023-11-15T11:22:05.669Z - [REQUEST]: 
[NX Daemon Server] - 2023-11-15T11:22:06.231Z - Time taken for 'hash changed files from watcher' 4.002299999818206ms
[NX Daemon Server] - 2023-11-15T11:22:06.260Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:23:03.836Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2023-11-15T11:23:03.837Z - Closed a connection. Number of open connections: 0
[NX Daemon Server] - 2023-11-15T11:23:03.838Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2023-11-15T11:23:03.839Z - [REQUEST]: Client Request for Project Graph Received
[NX Daemon Server] - 2023-11-15T11:23:03.840Z - [REQUEST]: Responding to the client. project-graph
[NX Daemon Server] - 2023-11-15T11:23:03.845Z - Time taken for 'total for creating and serializing project graph' 0.4910000003874302ms
[NX Daemon Server] - 2023-11-15T11:23:03.846Z - Done responding to the client project-graph
[NX Daemon Server] - 2023-11-15T11:23:03.846Z - Handled REQUEST_PROJECT_GRAPH. Handling time: 1. Response time: 6.
[NX Daemon Server] - 2023-11-15T11:23:04.159Z - [REQUEST]: Responding to the client. handleHashTasks
[NX Daemon Server] - 2023-11-15T11:23:04.160Z - Done responding to the client handleHashTasks
[NX Daemon Server] - 2023-11-15T11:23:04.160Z - Handled HASH_TASKS. Handling time: 206. Response time: 1.
[NX Daemon Server] - 2023-11-15T11:23:04.758Z - Closed a connection. Number of open connections: 0
[NX Daemon Server] - 2023-11-15T11:23:08.482Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2023-11-15T11:23:08.482Z - Closed a connection. Number of open connections: 0
[NX Daemon Server] - 2023-11-15T11:23:08.483Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2023-11-15T11:23:08.486Z - [REQUEST]: Client Request for Project Graph Received
[NX Daemon Server] - 2023-11-15T11:23:08.486Z - [REQUEST]: Responding to the client. project-graph
[NX Daemon Server] - 2023-11-15T11:23:08.491Z - Time taken for 'total for creating and serializing project graph' 0.20030000060796738ms
[NX Daemon Server] - 2023-11-15T11:23:08.492Z - Done responding to the client project-graph
[NX Daemon Server] - 2023-11-15T11:23:08.492Z - Handled REQUEST_PROJECT_GRAPH. Handling time: 0. Response time: 6.
[NX Daemon Server] - 2023-11-15T11:23:08.612Z - [REQUEST]: Responding to the client. handleHashTasks
[NX Daemon Server] - 2023-11-15T11:23:08.613Z - Done responding to the client handleHashTasks
[NX Daemon Server] - 2023-11-15T11:23:08.614Z - Handled HASH_TASKS. Handling time: 8. Response time: 2.
[NX Daemon Server] - 2023-11-15T11:23:09.085Z - [REQUEST]: Responding to the client. outputsHashesMatch
[NX Daemon Server] - 2023-11-15T11:23:09.085Z - Done responding to the client outputsHashesMatch
[NX Daemon Server] - 2023-11-15T11:23:09.085Z - Handled OUTPUTS_HASHES_MATCH. Handling time: 276. Response time: 0.
[NX Daemon Server] - 2023-11-15T11:23:09.155Z - [WATCHER]: 0 file(s) created or restored, 0 file(s) modified, 5 file(s) deleted
[NX Daemon Server] - 2023-11-15T11:23:09.156Z - [WATCHER]: Processing file changes in outputs
[NX Daemon Server] - 2023-11-15T11:23:09.156Z - [WATCHER]: Stopping the watcher for C:\Users\hanne\development\energy-assistant (sources)
[NX Daemon Server] - 2023-11-15T11:23:09.157Z - [WATCHER]: Stopping the watcher for C:\Users\hanne\development\energy-assistant (outputs)
[NX Daemon Server] - 2023-11-15T11:23:09.157Z - Server stopped because: "Stopping the daemon the set of ignored files changed (native)"

### Expected Behavior

The server should run and execute the task, at least list the files that cause the error in the log ^^, than I can maybe understand what the problem is ...

### Steps to Reproduce

1. Call the target
2. The server stops

### Nx Report

```shell
Node   : 18.16.0
   OS     : win32-x64
   pnpm   : 8.10.2

   nx                 : 17.0.3
   @nx/js             : 17.0.3
   @nx/jest           : 17.0.3
   @nx/linter         : 17.0.3
   @nx/eslint         : 17.0.3
   @nx/workspace      : 17.0.3
   @nx/angular        : 17.0.3
   @nx/cypress        : 17.0.3
   @nx/devkit         : 17.0.3
   @nx/eslint-plugin  : 17.0.3
   @nx/nest           : 17.0.3
   @nx/node           : 17.0.3
   @nx/storybook      : 17.0.3
   @nrwl/tao          : 17.0.3
   @nx/web            : 17.0.3
   @nx/webpack        : 17.0.3
   nx-cloud           : 16.5.2
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @compodoc/compodoc     : 1.1.22
   @ngrx/component-store  : 16.3.0
   @ngrx/effects          : 16.3.0
   @ngrx/entity           : 16.3.0
   @ngrx/router-store     : 16.3.0
   @ngrx/schematics       : 16.3.0
   @ngrx/store            : 16.3.0
   @ngrx/store-devtools   : 16.3.0
   @nx-tools/nx-container : 5.1.0
   @storybook/angular     : 7.5.3
   ng-mocks               : 14.11.0
```

### Package Manager Version

pnpm 8.10

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by AgentEnder** [comment]
The set of ignore files changing indicates that a new .gitignore or .nxignore file was located, is it possible that one of your targets is copying an ignore file?

**Comment by roderik** [comment]
I have a similar issue, i have a codegen step that creates a gitignored file. But i need to copy it over in assets in a build step so i unignore it in .nxignore. So every time this happens, it shuts down.

**Comment by MaxKless** [comment]
Hey! I'm closing this as a duplicate of https://github.com/nrwl/nx/issues/16718.
I believe this issue is resolved in the latest version of Nx. If you still experience problems here, please report them with a reproduction on the linked issue.
